helm/provider.go

```diff
@@ -81,7 +84,12 @@ func Provider() terraform.ResourceProvider {
 							return
 						}
 					}
-					errs = append(errs, fmt.Errorf("%s must be a valid storage driver", v))
+					diags = append(diags, diag.Diagnostic{
```

Comment:
I understand. I found it confusing too! I had to read the example a few times before I realized that somehow `diags` just magically came into existence without being declared explicitly. :) https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-diagnostics

I imagine other providers will copy from this example too. But I'm fine with either implementation. Do you still prefer to explicitly create the diag.Diagnostics here?

Reply:
I think for their example it actually does make sense, because they're appending multiple diagnostics to the slice.

I think for our case we can be explicit and get rid of the naked returns.